packages/fxa-payments-server/src/lib/apiClient.ts

```diff
@@ -412,10 +414,6 @@ export async function apiRetrieveCouponDetails(params: {
       { body: JSON.stringify(params) }
     );
 
-    if (!result.discountAmount) {
-      throw new Error(CouponErrorMessageType.Generic);
-    }
-
     Amplitude.coupon_FULFILLED(metricsOptions);
```

Comment:
Now that we don't throw in this function anymore, maybe we should clarify whether this metric mean successful request or valid coupon?

Reply:
The VPN Coupons MVP doc says the following.

```
coupon_error: coupon validation failed. Capture failure type with coupon code used
coupon_success: fire when user presses apply and coupon is valid

```

It sounds like only fire the success event when the coupon is actually valid.

As part of these changes, I moved all the coupon metric events to the CouponForm component, since that's where the logic is at the moment that determines whether or not a coupon is valid.